core/src/server/rpc_module.rs

```diff
@@ -923,14 +899,29 @@ impl SubscriptionSink {
 		T: Serialize,
 		E: std::fmt::Display,
 	{
+		if let Err(SubscriptionAcceptRejectError::RemotePeerAborted) = self.accept() {
+			return SubscriptionClosed::RemotePeerAborted;
+		}
+
 		let conn_closed = match self.close_notify.as_ref().map(|cn| cn.handle()) {
 			Some(cn) => cn,
 			None => {
 				return SubscriptionClosed::RemotePeerAborted;
 			}
 		};
 
-		let mut sub_closed = self.unsubscribe.clone();
+		let mut sub_closed = match self.unsubscribe.as_ref() {
+			Some(rx) => rx.clone(),
+			_ => {
```

Comment:
style nit: `_ => return ...` would help me understand this is short-circuiting, at a glance